tests/geospatial/test_zonal_average.py

```diff
@@ -0,0 +1,58 @@
+"""
+This example was adapted from https://github.com/dcherian/dask-demo/blob/main/nwm-aws.ipynb
+"""
+
+import flox.xarray
+import numpy as np
+import rioxarray
+import xarray as xr
+
+
+def test_nwm(
+    s3,
+    scale,
+    client_factory,
+    cluster_kwargs={
+        "workspace": "dask-engineering",
+        "region": "us-east-1",
+        "wait_for_workers": True,
+    },
+    scale_kwargs={
+        "small": {"n_workers": 10},
+        "large": {"n_workers": 200, "scheduler_memory": "32 GiB"},
+    },
+):
+    with client_factory(
+        **scale_kwargs[scale], **cluster_kwargs
+    ) as client:  # noqa: F841
+        ds = xr.open_zarr(
+            "s3://noaa-nwm-retrospective-2-1-zarr-pds/rtout.zarr", consolidated=True
+        )
+
+        if scale == "small":
+            # 6.03 TiB
+            time_range = slice("2020-01-01", "2020-12-31")
+        else:
+            # 252.30 TiB
+            time_range = slice("1979-02-01", "2020-12-31")
+        subset = ds.zwattablrt.sel(time=time_range)
+
+        counties = rioxarray.open_rasterio(
+            s3.open("s3://nwm-250m-us-counties/Counties_on_250m_grid.tif"),
```

Comment:
You could parameterize on two files here:
```
            s3.open("s3://nwm-250m-us-counties/HUC12s_on_1000m_grid.tif"),
```
this is 100_000 watersheds instead of 3000 counties

Reply:
Do you have the path for the other file?

Reply:
Ah, I take it it's `s3://nwm-250m-us-counties/HUC12s_on_1000m_grid.tif`?

Reply:
ah actually this isn't on the right grid yet. I'll get back to you on this.

Reply:
OK here you go: `"s3://nwm-250m-us-counties/HUC12s_on_250m_grid.tif"` It *should* slot right in.

Reply:
Awesome, thanks @dcherian. I'm going to circle back to this as I'd like to merge this PR in so I can make use of some of the fixtures here in another PR